### PR TITLE
REDBOX-440 - Editable chat title

### PR DESCRIPTION
**Pull request by brunns**
## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests

### CODE REVIEW

File: tests/test_journey.py [hunk 1]
```diff
@@ -113,6 +113,13 @@ def test_user_journey(page: Page, email_address: str):
         latest_chat_response = chats_page.wait_for_latest_message()
         assert latest_chat_response.text
         assert latest_chat_response.route.startswith(route)
+
+    # Navigate to old chat & rename
+    chats_page = chats_page.navigate_to_titled_chat("What architecture is in use?")
+    chats_page.chat_title = "About tech stuff."
+    chats_page = chats_page.start_new_chat()
+    chats_page = chats_page.navigate_to_titled_chat("About tech stuff.")
```
Comment (rachaelcodes) [inline tests/test_journey.py:121]: I really like the tests - it's v clear what you've changed

File: django_app/frontend/src/js/chats/streaming.js [hunk 2]
```diff
@@ -297,3 +307,125 @@ class DocumentSelector extends HTMLElement {
   }
 }
 customElements.define("document-selector", DocumentSelector);
+
```
Comment (rachaelcodes) [inline django_app/frontend/src/js/chats/streaming.js:310]: Question: do we want to start thinking about breaking this file up into its different components?

Comment (brunns) [reply django_app/frontend/src/js/chats/streaming.js:310]: That's not a bad idea at all. The feedback component is already in a file of its own.

File: django_app/frontend/src/js/chats/streaming.js [hunk 3]
```diff
@@ -297,3 +307,125 @@ class DocumentSelector extends HTMLElement {
...
+    if (this.input && this.heading) {
+      this.input.value = e.detail.title;
+      this.heading.textContent = `${e.detail.title}`;
+      this.heading.classList.remove("govuk-visually-hidden");
+    }
+  };
+
+  update = () => {
+    const newTitle = this.input?.value;
+    console.log(`updating chat title to "${newTitle}"`);
```
Comment (rachaelcodes) [inline django_app/frontend/src/js/chats/streaming.js:409]: are we keeping this for the user?

Comment (brunns) [reply django_app/frontend/src/js/chats/streaming.js:409]: Good catch - I'll get rid.

Comment (brunns) [reply django_app/frontend/src/js/chats/streaming.js]: Gone.

File: django_app/frontend/src/js/chats/streaming.js (hunk 2 above)
Comment (KevinEtchells) [reply django_app/frontend/src/js/chats/streaming.js]: I think that's a good idea, and now we have Parcel doing the bundling this should be relatively clean and simple.

**Pull request merged**